src/Aserto.AspNetCore.Middleware/Clients/Directory/DirectoryAPIClient.cs

```diff
@@ -0,0 +1,276 @@
+    using Aserto.Directory.Reader.V2;
+    using Aserto.Directory.Writer.V2;
+    using Google.Protobuf.WellKnownTypes;
+    using Grpc.Net.Client;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.Extensions.Logging;
+    using Microsoft.Extensions.Options;
+    using static Aserto.Directory.Reader.V2.Reader;
+    using static Aserto.Directory.Writer.V2.Writer;
+
+    /// <summary>
+    /// Client for Aserto Directory API.
+    /// </summary>
+    public class DirectoryAPIClient : IDirectoryAPIClient
+    {
+        private readonly ReaderClient readerClient;
+        private readonly WriterClient writerClient;
+        private readonly Grpc.Core.Metadata metaData;
+        private readonly AsertoDirectoryOptions options;
+        private readonly ILogger logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DirectoryAPIClient"/> class.
+        /// </summary>
+        /// <param name="options">Authorizer API Client options <see cref="AsertoDirectoryOptions"/>.</param>
+        /// <param name="loggerFactory">The <see cref="ILoggerFactory"/> for this class.</param>
+        public DirectoryAPIClient(IOptions<AsertoDirectoryOptions> options, ILoggerFactory loggerFactory)
+        {
+            this.logger = loggerFactory.CreateLogger<DirectoryAPIClient>();
+
+            if (options != null)
+            {
+                    this.options = options.Value;
+            }
+            else
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            if (!AsertoDirectoryOptions.Validate(this.options))
+            {
+                throw new ArgumentException("wrong url provided for DirectoryServiceUrl");
+            }
+
+            var grpcChannelOptions = new GrpcChannelOptions { };
+
+            if (this.options.Insecure)
+            {
+                var httpHandler = new HttpClientHandler
+                {
+                    ServerCertificateCustomValidationCallback =
+                HttpClientHandler.DangerousAcceptAnyServerCertificateValidator,
+                };
+
+                grpcChannelOptions = new GrpcChannelOptions { HttpHandler = httpHandler };
+            }
+
+            var channel = GrpcChannel.ForAddress(
+                this.options.DirectoryServiceUrl,
+                grpcChannelOptions);
+
+            this.readerClient = new ReaderClient(channel);
+            this.writerClient = new WriterClient(channel);
+
+            this.metaData = new Grpc.Core.Metadata
+            {
+                { "Aserto-Tenant-Id", $"{this.options.TenantID}" },
+                { "Authorization", $"basic {this.options.DirectoryApiKey}" },
+            };
+        }
+
+        /// <inheritdoc/>
+        public async Task<GetObjectTypeResponse> GetObjectTypeAsync(GetObjectTypeRequest req)
+        {
+            var result = await this.readerClient.GetObjectTypeAsync(req, this.metaData);
```

Comment:
we can use an [Interceptor](https://learn.microsoft.com/en-us/aspnet/core/grpc/interceptors?view=aspnetcore-7.0) to avoid passing metadata to each request, in this way adding new methods we do not need to take care of metadata.